future-apps/payroll/contracts/Payroll.sol

```diff
@@ -606,8 +603,18 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
      */
     function _getLastPayroll(uint256 _employeeId, uint256 _payedAmount) internal view returns (uint64) {
         Employee storage employee = employees[_employeeId];
-        uint64 timeDiff = _payedAmount.div(employee.denominationTokenSalary).toUint64();
-        return employee.lastPayroll.add(timeDiff);
+
+        uint64 timeDiff = uint64(_payedAmount.div(employee.denominationTokenSalary));
+
+        // This function is only called from _payday, where we make sure that _payedAmount is lower or equal to the
+        // total owed amount, that is obtained from _getOwedSalary, which does exactly the opposite calculation:
+        // multiplying the employee's salary by an uint64 number of seconds. Therefore, timeDiff will always fit in 64.
+        // Nevertheless, we are performing a sanity check at the end to ensure the computed last payroll timestamp
+        // is not greater than the current timestamp.
+
+        uint64 lastPayrollDate = employee.lastPayroll.add(timeDiff);
```

Comment:
Nit pick: wouldn't it be better to do all the math on `uint256`, do the check, and then finally cast it to `uint64` at the end? As we have discussed, there's no way this can overflow as it is right now, but just in case, to be safer.